### PR TITLE
add multinomial op

### PR DESCRIPTION
**Pull request by pangyoki**
### PR types

New features

### PR changes

OPs

### Describe

- add `multinomial` op (C++ op)
- add `paddle.multinomial(x, num_samples=1, replacement=False, name=None)` (python API)

    - x(Tensor):  A tensor with probabilities for generating the random number.
    - num_samples(int, optional): Number of samples, default is 1.
    - replacement(bool, optional): Whether it is a replaceable sample, default is False. If replacement is False, it's not a replaceable sample. Every category can be used only once.

### CODE REVIEW

File: python/paddle/tensor/random.py [hunk 1]
```diff
@@ -85,6 +86,71 @@ def bernoulli(x, name=None):
...
+            refer to :ref:`api_guide_Name`.
+    Returns:
+        Tensor: A Tensor filled with sampled category index after ``num_samples`` times samples.
+
+    Examples:
+        .. code-block:: python
+
+        import paddle
+
+        paddle.disable_static()
```
Comment (zhiqiu) [inline python/paddle/tensor/random.py:113]: It can be removed, you can refine it in the next PR.

**Review by zhiqiu — APPROVED** [review]
LGTM

**Review by kolinwei — APPROVED** [review]

**Review by jzhang533 — APPROVED** [review]
lgtm

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

File: python/paddle/tensor/random.py (hunk 1 above)
Comment (pangyoki) [reply python/paddle/tensor/random.py:113]: Done, refer to PR https://github.com/PaddlePaddle/Paddle/pull/27736

**Pull request merged**